narwhal/primary/src/header_waiter.rs

```diff
@@ -330,9 +339,12 @@ impl HeaderWaiter {
                         },
                         Ok(header) => {
                             self.cleanup_pending_requests(&header);
-
                             // Ok to drop the header if core is overloaded.
-                            let _ = self.tx_headers_loopback.try_send(header);
+                            if let Err(e) = self.tx_headers_loopback.try_send(header.clone()) {
+                                warn!("Failed to loop back ready header {header} {e}");
+                            } else {
+                                debug!("Header looped back to core: {header}");
```

Comment:
Maybe that's a bit noisy , I mean under normal circumstances we now that this will succeed

Reply:
Yeah maybe there are already logs in the loopback handler, otherwise this is useful as an indicator for cases of deadlocks.